File: locaway/src/main/java/com/sopra/LocAway/model/Bookmark.java
package com.sopra.LocAway.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
public class Bookmark {
	@Id
	@GeneratedValue
	@JsonView(Views.ViewCommon.class)
	private Long id;
	@Version
	@JsonView(Views.ViewCommon.class)
	private int version;
	@ManyToOne
	@JsonView(Views.ViewBookmark.class)
	private User user;
	@ManyToOne
	@JsonView(Views.ViewBookmark.class)
	private Accomodation accomodation;

	public Bookmark() {
	}

	public Bookmark(User user, Accomodation accomodation) {
		super();
		this.user = user;
		this.accomodation = accomodation;
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public Accomodation getAccomodation() {
		return accomodation;
	}

	public void setAccomodation(Accomodation accomodation) {
		this.accomodation = accomodation;
	}

}

File: LocAwayAngular/src/app/Model/Bookmark.ts
import {User} from "./User";
import {Accomodation} from "./Accomodation";

export class Bookmark{
  public id: number;
  public version: number;
  public user: User;
  public accomodation: Accomodation;


  constructor(id: number, version: number, user: User, accomodation: Accomodation) {
    this.id = id;
    this.version = version;
    this.user = user;
    this.accomodation = accomodation;
  }
}
